### PR TITLE
[WFLY-18882] Add resource adapter subsystem dependencies based on cap…

### PR DESCRIPTION
**Pull request by tadamski**
…ability

____

<--- THIS SECTION IS AUTOMATICALLY GENERATED BY WILDFLY GITHUB BOT. ANY MANUAL CHANGES WILL BE LOST. --->

> Wildfly issue links:
> * [WFLY-14266](https://issues.redhat.com/browse/WFLY-14266)
> * [WFLY-18882](https://issues.redhat.com/browse/WFLY-18882)

<--- END OF WILDFLY GITHUB BOT REPORT --->

More information about the [wildfly-bot[bot]](https://github.com/wildfly/wildfly-github-bot)

### CODE REVIEW

**Review by darranl — DISMISSED** [review]
@tadamski Is the correct Jira issue referenced?  This seems to be about adding some capability references but the referenced Jira issue it about adding new config?

This PR looks to have failed test runs associated.

You should have the permissions to be able to request some reviews of the changes so we can see if they can be merged without handling as an RFE.

**Comment by tadamski** [comment]
@darranl I think the issue is correct. IIRC there was an error in appclient tests which tried to extract property from nonexistent subsystem. I proposed this fix https://github.com/wildfly/wildfly/pull/15418, but from the discussion it turned out that it would better to use capabilitie hence this PR.

I have rebased it and cannot reproduce the error neither on JDK17 nor JDK11.

**Comment by tadamski** [comment]
now there is some other test failing but tbh I don't expect it to be related

File: connector/src/main/java/org/jboss/as/connector/deployers/ra/Capabilities.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
```
Comment (darranl) [inline connector/src/main/java/org/jboss/as/connector/deployers/ra/Capabilities.java:21]: Please update to the new license header and use the Apache licence.

e.g. https://github.com/wildfly/wildfly/blob/main/connector/src/main/java/org/jboss/as/connector/deployers/ra/RaDeploymentActivator.java#L1-L4

**Review by darranl — CHANGES_REQUESTED** [review]
Small licence change needed.

**Comment by darranl** [comment]
@pferraro Would you be able to please help with a review of the capability handling here.

**Comment by pferraro** [comment]
@darranl The logic in this PR looks sound to me (given my note below), however, the linked jira is a feature request, whereas, the proposed change here is clearly a bug fix.  Do we need a separate jira for this?
@tadamski N.B. Normally, your "org.wildfly.resource-adapters" would expose the value provided by the service installed via the resource-adapters subsystem add operation - however, that service does not sufficiently encapsulate its provided value, so we definitely do not want this exposed via a public contract.

**Comment by tadamski** [comment]
This is a bug fix in an RFE that was already merged so I used WFLY jira. I can create bug specific one if necessary.

**Comment by pferraro** [comment]
@tadamski I see. Since it looks like WFLY-14266 was merged several releases ago, I think a separate bug fix jira (targeting 31.0.0.Final) is warranted.

**Comment by tadamski** [comment]
@pferraro I have created WFLY-18882 and opened https://github.com/wildfly/wildfly/pull/17512

**Comment by bstansberry** [comment]
@tadamski Please rebase and change the commit message to WFLY-18882

**Review by pferraro — APPROVED** [review]

**Comment by tadamski** [comment]
@bstansberry done

File: connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ResourceAdaptersRootResourceDefinition.java [hunk 2]
```diff
@@ -18,6 +19,10 @@
  */
 public class ResourceAdaptersRootResourceDefinition extends SimpleResourceDefinition {
 
+    private static final String RESOURCE_ADAPTERS_SUBSYSTEM_CAPABILITY = "org.wildfly.resource-adapters";
+
+    static final RuntimeCapability<Void> RESOURCE_ADAPTERS_SUBSYSTEM = RuntimeCapability.Builder.of(RESOURCE_ADAPTERS_SUBSYSTEM_CAPABILITY).build();
```
[the remark this answers is not in the document]
Comment (tadamski) [reply connector/src/main/java/org/jboss/as/connector/subsystems/resourceadapters/ResourceAdaptersRootResourceDefinition.java]: fixed

**Comment by tadamski** [comment]
@bstansberry rebased but apparently this was fixed in other PR and after rebase it is just access and naming change

**Comment by bstansberry** [comment]
Superseded by #17512

**Pull request closed without merging**